core/src/common.cpp

```diff
@@ -237,6 +385,30 @@ namespace irods::http
 		// Verify the bearer token is known to the server. If not, return an error.
 		auto mapped_value{irods::http::process_stash::find(bearer_token)};
 		if (!mapped_value.has_value()) {
+			// If we're running as a protected resource, assume we have a OIDC token
+			if (irods::http::globals::oidc_configuration().at("run_as_protected_resource").get<bool>()) {
+				body_arguments args{
+					{"token", bearer_token},
+					{"token_type_hint", "access_token"}};
+
+				auto json_res{hit_introspection_endpoint(url_encode_body(args))};
+
+				// Validate access token
+				if (!json_res.at("active").get<bool>()) {
+					log::warn("{}: Access token is invalid or expired.", __func__);
+					return {.response = fail(status_type::unauthorized)};
+				}
+
+				// Do mapping of user to irods user
+				auto user{map_json_to_user(json_res)};
+				if (user) {
+					return {.client_info = {.username = *std::move(user)}};
+				}
+
+				log::error("{}: Could not find a matching user.", __func__);
```

Comment:
Is there anything worth we can/should include about the user to help the admin here?

Reply:
We could say "check either irods_user_claim or user attribute mapping", but I think that might not be best?

Just because this may actually be a common error (i.e. non-irods/non-mapped user trying to access the endpoint).
Might be worth going from 'error' to 'warn'.

Reply:
I think I agree. The current message is good/fine.

How do you feel about logging a substring of the bearer token?

Also, logging as a warning sounds good.

Reply:
what would the (partial) token tell the admin that is helpful?   the access token is time-bound / ephemeral, right?

doesn't feel good to log it on first glance...

Reply:
We don't have to handle it now. We can change things later.